src/lib.rs

```diff
@@ -98,6 +98,27 @@ impl<T> Arena<T> {
     pub fn is_empty(&self) -> bool {
         self.count() == 0
     }
+
+    #[inline]
+    pub fn get(&self, id: NodeId) -> Option<&Node<T>> {
+        self.nodes.get(id.index)
+    }
+
+    #[inline]
+    pub fn get_mut(&mut self, id: NodeId) -> Option<&mut Node<T>> {
+        self.nodes.get_mut(id.index)
+    }
+
+    pub fn find<P>(&mut self, predicate: P) -> Option<&Node<T>>
```

Comment:
So I think it would be a bit more comfortable to just add an `iter` API like this:

```rust
pub fn iter(&self) -> std::slice::Iter<Node<T>> {
    self.nodes.iter()
}
```

Please add documentation and testing too.